File: bench/json.rb
#!/usr/bin/env ruby
# frozen_string_literal: true
# JSON round-trip - json module (stdlib since Ruby 1.9)
require 'json'

n = 5000
objs = []
(1..n).each do |i|
  objs << { id: i, name: "user-#{i}", active: i % 3 == 0, score: i * 1.5, tags: %w[a b c] }
end

start = Process.clock_gettime(Process::CLOCK_MONOTONIC)
text = objs.to_json
encoded = (Process.clock_gettime(Process::CLOCK_MONOTONIC) - start) * 1000

start = Process.clock_gettime(Process::CLOCK_MONOTONIC)
back = JSON.parse(text)
decoded = (Process.clock_gettime(Process::CLOCK_MONOTONIC) - start) * 1000

puts "JSON encode #{n} objects in #{encoded}ms, decode in #{decoded}ms"

File: bench/json.js
// JSON round-trip - JSON.parse/stringify (stdlib)
const n = 5000;
const objs = [];
for (let i = 1; i <= n; i++) {
  objs.push({ id: i, name: "user-" + i, active: i % 3 === 0, score: i * 1.5, tags: ["a", "b", "c"] });
}

let start = process.hrtime.bigint();
const text = JSON.stringify(objs);
const encoded = Number(process.hrtime.bigint() - start) / 1e6;

start = process.hrtime.bigint();
const back = JSON.parse(text);
const decoded = Number(process.hrtime.bigint() - start) / 1e6;

console.log(`JSON encode ${n} objects in ${encoded}ms, decode in ${decoded}ms`);
